### PR TITLE
Adding must-gather for oadp

### PR DESCRIPTION
**Pull request by savitharaghunathan**
Adding must-gather for oadp

### CODE REVIEW

File: must-gather/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# OADP must-gather
+
+`must-gather` is a tool built on top of [OpenShift must-gather](https://github.com/openshift/must-gather)
+that expands its capabilities to gather OADP operator specific resources
+
+### Usage
+
+**Full gather**
+```sh
+oc adm must-gather --image=quay.io/oadp/must-gather:latest
```
Comment (savitharaghunathan) [inline must-gather/README.md:10]: Do we need to configure `quay.io/oadp/must-gather` repo in `quay.io` ?

File: must-gather/collection-scripts/gather [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+# Default to gathering 72h of logs unless specified as var
+logs_since="${logs_since:-72h}"
+
+unset KUBECONFIG
+
+clusterID=$(oc get clusterversion -o jsonpath='{.items[].spec.clusterID}' | cut -c -8)
+namespaces=()
+#to verify: one oadp CR per cluster or multiple CRs per clsuter(since it is namespace scoped)
+for ns in $(oc get veleroes.oadp.openshift.io --all-namespaces --no-headers | awk '{print $1}')
```
Comment (dymurray) [inline must-gather/collection-scripts/gather:11]: `dataprotectionapplications.oadp.openshift.io`

File: must-gather/collection-scripts/gather_crs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      object_collection_path=/must-gather/clusters/${clusterID}/namespaces/${ocproject}/velero.io/${resource}
+      mkdir -p ${object_collection_path}
+      /usr/bin/oc get ${resource} -n ${ocproject} -o yaml ${ocobject} &> ${object_collection_path}/${ocobject}.yaml & 
+    fi
+    pwait $max_parallelism
+  done
+done
+
+for ns in ${namespaces[@]}; do
+  for vel in $(oc get velero --namespace ${ns} --no-headers | awk '{print $1}'); do
```
Comment (dymurray) [inline must-gather/collection-scripts/gather_crs:51]: `oc get dpa`

File: must-gather/collection-scripts/gather_crs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  while read ocresource; do
+    ocobject=$(echo $ocresource | awk '{print $1}')
+    ocproject=$(echo $ocresource | awk '{print $2}')
+    if [ -z "${ocproject}" ]|[ "${ocproject}" == "<none>" ]; then
+      object_collection_path=/must-gather/clusters/${clusterID}/cluster-scoped-resources/${resource}
+      mkdir -p ${object_collection_path}
+      /usr/bin/oc get ${resource} -o yaml ${ocobject} &> ${object_collection_path}/${ocobject}.yaml &
+    else
+      #TODO: verify if there are more crds to add for oadp
+      skip=("veleroes.oadp.openshift.io")
```
Comment (dymurray) [inline must-gather/collection-scripts/gather_crs:38]: `dataprotectionapplications.oadp.openshift.io`

File: must-gather/collection-scripts/gather_crs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      /usr/bin/oc get ${resource} -n ${ocproject} -o yaml ${ocobject} &> ${object_collection_path}/${ocobject}.yaml & 
+    fi
+    pwait $max_parallelism
+  done
+done
+
+for ns in ${namespaces[@]}; do
+  for vel in $(oc get velero --namespace ${ns} --no-headers | awk '{print $1}'); do
+    mkdir -p "/must-gather/clusters/${clusterID}/namespaces/${ns}/oadp.openshift.io/dpa-${vel}"
+    oc get velero ${vel} -o yaml --namespace ${ns} &> "/must-gather/clusters/${clusterID}/namespaces/${ns}/oadp.openshift.io/dpa-${vel}/${vel}.yml" &
```
Comment (dymurray) [inline must-gather/collection-scripts/gather_crs:53]: `oc get dpa`

File: must-gather/collection-scripts/gather_logs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+namespaces=("$@")
+
+
+# Collect all Pod logs from namespaces where OADP operator is installed
+for ns in ${namespaces[@]}; do
+  # Gather Pod associated logs
+  /usr/bin/gather_logs_pods ${cluster} ${ns} ${logs_since} ${max_parallelism} &
+
+  # Gather logs for each oadp operator instance
+  for vel in $(oc get velero --namespace ${ns} --no-headers | awk '{print $1}'); do
```
Comment (dymurray) [inline must-gather/collection-scripts/gather_logs:19]: `oc get dpa`

File: must-gather/README.md (hunk 1 above)
Comment (dymurray) [reply must-gather/README.md:10]: @savitharaghunathan please change this to `quay.io/konveyor/must-gather`

Comment (savitharaghunathan) [reply must-gather/README.md:10]: @dymurray That would point to MTC must-gather right?

Comment (savitharaghunathan) [reply must-gather/README.md:10]: Sorry. I missed the oadp prefix in there. Fix coming up right away

Comment (savitharaghunathan) [reply must-gather/README.md]: done 👍

**Review by dymurray — APPROVED** [review]

**Pull request merged**